python/lsst/analysis/tools/tasks/base.py

```diff
@@ -305,6 +327,7 @@ def loadData(self, handle: DeferredDatasetHandle, names: Iterable[str] | None =
         """
         if names is None:
             names = self.collectInputNames()
+        names.add("patch")
```

Comment:
Why not just put this line in `collectInputNames`? This would also prevent the mypy error that you are getting because not every `Iterable` has an `add` method.

Reply:
This was left over from when I was experimenting with something. I've removed it and everything seems to be working.